machineconfiguration/v1alpha1/types_machineosbuild.go

```diff
@@ -0,0 +1,155 @@
+// +openshift:compatibility-gen:level=4
+type MachineOSBuild struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec describes the configuration of the machine os build
+	// +kubebuilder:validation:Required
+	Spec MachineOSBuildSpec `json:"spec"`
+
+	// status describes the lst observed state of this machine os build
+	// +optional
+	Status MachineOSBuildStatus `json:"status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// MachineOSBuildList describes all of the Builds on the system
+//
+// Compatibility level 4: No compatibility is provided, the API can change at any point for any reason. These capabilities should not be used by applications needing long term support.
+// +openshift:compatibility-gen:level=4
+type MachineOSBuildList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []MachineOSBuild `json:"items"`
+}
+
+// MachineOSBuildSpec describes information about a build process primarily populated from a MachineOSConfig object.
+type MachineOSBuildSpec struct {
+	// configGeneration tracks which version of MachineOSConfig this build is based off of
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Required
+	ConfigGeneration int64 `json:"configGeneration"`
+	// desiredConfig is the desired config we want to build an image for.
+	// +kubebuilder:validation:Required
+	DesiredConfig RenderedMachineConfigReference `json:"desiredConfig"`
+	// machineOSConfig is the config object which the build is based off of
+	// +kubebuilder:validation:Required
+	MachineOSConfig MachineOSConfigReference `json:"machineOSConfig"`
+	// version tracks the newest MachineOSBuild for each MachineOSConfig
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Required
+	Version int64 `json:"version"`
+}
+
+// MachineOSBuildStatus describes the state of a build and other helpful information.
+// +union
+// +kubebuilder:validation:XValidation:rule="has(self.imageBuilderType) && self.imageBuilderType == 'PodImageBuilder' ?  true : !has(self.buildPod)",message="buildPod is required when imageBuilderType is PodImageBuilder, and forbidden otherwise"
+type MachineOSBuildStatus struct {
+	// conditions are state related conditions for the build. Valid types are:
+	// Prepared, Building, Failed, Interrupted, Restarted, and Ready
+	// once a Build is marked as Failed, no future conditions can be set. This is enforced by the MCO.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	// ImageBuilderType describes the image builder set in the MachineOSConfig
+	// +unionDiscriminator
+	ImageBuilderType MachineOSImageBuilderType `json:"imageBuilderType"`
+	// relatedObjects is a list of objects that are related to the build process.
```

Comment:
```go
type MachineOSBuildStatus struct {
	RelatedObjects []ObjectReference `json:"relatedObjects,omitempty"`
}
```

you can put things like
1. namespaces
2. direct configmaps (I was configured from here)
3. MachineOSConfig
4. MachineConfigPool